specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-10-01-preview/SourceControls.json

```diff
@@ -959,11 +998,7 @@
         "required": [
           "repositoryAccess"
         ],
-        "properties": {
-          "repositoryAccess": {
-            "$ref": "#/definitions/RepositoryAccess"
-          }
-        }
+        "$ref": "#/definitions/RepositoryAccessProperties"
```

Comment:
>RepositoryAccessProperties [](http://example.com/codeflow?start=31&length=26)

this seems like a circular dependency right ?  The RepositoryAccess refers to RepositoryAccessProperties which refers to RepositoryAccessObject which again refers to RepositoryAccess .

Reply:
No, it is not. Chain is the following: `#/parameters/RepositoryAccess` -> `#/definitions/RepositoryAccessProperties` -> `#/definitions/RepositoryAccessObject` -> `#/definitions/RepositoryAccess`